sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -126,6 +127,80 @@ static pi_result getExtFuncFromContext(pi_context context, T *fptr) {
   return cast<pi_result>(ret_err);
 }
 
+// This is a temporary workaround for a clMemBlockingFreeINTEL bug in OpenCL CPU
+// runtime. Chooses clMemBlockingFreeINTEL for GPU and clMemFreeINTEL for other
```

Comment:
Please explain in comment what is the intended behavior, and why it is not always possible.

Reply:
I've expanded the comment.